src/plugins/workspace/server/workspace_client.ts

```diff
@@ -110,6 +115,12 @@ export class WorkspaceClient implements IWorkspaceClientImpl {
           permissions,
         }
       );
+      if (dataSources) {
```

Comment:
Shall we assign dataSources before `create` the workspace?

Reply:
I think this should be added once we figure out the logic of default DS in workspace. It would be added in a separate PR.

Reply:
Yes we can, also i'm thinking whether we should add a logic that call deleteFromWorkspaces if creation is failed, what do you think?

Reply:
That would be better, for now I think adjusting the order should be good enough.